contracts/v0.2/src/SubmissionProxy.sol

```diff
@@ -15,25 +15,63 @@ contract SubmissionProxy is Ownable {
     uint256 public maxSubmission = 50;
     uint256 public expirationPeriod = 5 weeks;
     mapping(address oracle => uint256 expiration) oracles;
+    mapping(string => address) feeds;
+    address[] public feedAddresses;
```

Comment:
How about removing `feedAddresses`? 
The only usage of feedAddress is when calling `getFeeds()` yet I'm not sure if we really need to know all feeds inside submissionProxy.

It's a good to have but this iteration is not something which is required in service perspective. if it's not something which is must have feature for the service such as giving yields to all clients in mapping, this can be all checked through calling from offchain scripts.

Reply:
The main purpose for the `feedAddresses` is to have access to instant knowledge about registered feeds. It could be accessed by parsing and keeping track of emitted events, but such approach is far from convenient. `SubmissionProxy` is a crucial component which have direct access to submission of data to feeds. I am assuming we won't have only a single `SubmissionProxy` but multiple of them, therefore we will have to keep track of them to know which one is responsible for set of feeds, and additionally, we don't want to keep hanging permission for some outdated data feeds.

Reply:
- list of feed pairs will be kept in orakl-config repo, and if feeds mappingis public, `feeds(${pair_name})` can be called to get feed contract. then it won't have to be accessed through emitted events
- I couldn't get the meaning of hanging permission, but the size of mapping won't effect that much on gas for accessing mapping data. how about just keeping the outdated datafeeds in mapping?

Reply:
* It possible to get values by passing keys to mappings, but you can't be sure you have all the keys, especially when dealing with large number of them. Making sure we know exactly what is inside is worth is small price for a bit of  more code.
* If we split batch submissions to sub-batch submissions (meaning that not all data feeds are going to be submitted at the same time) we don't need to have multiple `SubmissionProxy` in order to keep the gas per single transaction low. I do not worry about using an extra gas for having data in mapping, but we should not allow to submit to old `Feed` contracts that are not meant to be used, to keep security perimeter under control.

Reply:
- I see, since we keep track of feeds in orakl-config, and as accessing through feed name, I assumed it'd be hard to miss out. But who knows, there could be a typo or some mistake.
- I see, I wasn't worrying about submitting to wrong feed contract since feed names are all tracked through orakl-config, it means feed addresses could be checked through multiple calling to the feeds mapping. but still, keeping it safer than keeping code smaller would be more important